apps/gateway/src/common/messageSchema.ts

```diff
@@ -32,13 +32,22 @@ export const messageSchema = z
     z.object({
       role: z.literal(MessageRole.user),
       name: z.string().optional(),
-      content: contentSchema,
+      content: userContentSchema,
     }),
   )
   .or(
     z.object({
       role: z.literal(MessageRole.assistant),
-      content: z.string(),
+      content: z.string().or(
+        z.array(
+          z.object({
+            type: z.literal(ContentType.toolCall),
+            toolCallId: z.string(),
+            toolName: z.string(),
+            args: z.record(z.any()),
+          }),
+        ),
+      ),
       toolCalls: z.array(
         z.object({
           id: z.string(),
```

Comment:
`toolCalls` is not part of the `content`, is it? An assistant message could contain both a set of `toolCalls` and a `content` string.

Reply:
nein, it can also contain an array of tool calls within the content in fact

Reply:
At least according to openai schema. Is vercel's ai different?

Reply:
yes it's different